src/keystore.c

```diff
@@ -427,6 +427,20 @@ static bool _get_xprv(const uint32_t* keypath, const size_t keypath_len, struct
     return true;
 }
 
+bool get_root_fingerprint(char* fingerprint, int fingerprint_len)
+{
+    static const uint32_t keypath[] = {
+        0 + BIP32_INITIAL_HARDENED_CHILD,
+    };
+
+    struct ext_key derived_xpub __attribute__((__cleanup__(keystore_zero_xkey))) = {0};
+    if (!keystore_get_xpub(keypath, sizeof(keypath) / sizeof(uint32_t), &derived_xpub)) {
+        return false;
+    }
+    snprintf(fingerprint, fingerprint_len, "%s", derived_xpub.hash160);
```

Comment:
hash160 is a bytes array, not a string, it is meaningless to use snprintf here and null terminate it. This will even cut your 4 bytes fingerprint to 3 bytes + `\0`.

Use `memcpy` instead to copy the first 4 bytes.